src/search/product.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * The structure of a product.
+ */
+export interface Product extends Record<string, any> {
```

Comment:
Does this need to extend `Record`?

Reply:
Yes it does, extending Record allows us to create whatever key and value pairs that we want and allows users to add more key/value pairs if they need them to their Product objects.